pkg/mimir/modules.go

```diff
@@ -379,6 +380,9 @@ func (t *Mimir) initTenantFederation() (serv services.Service, err error) {
 //	                                          │                  │
 //	                                          └──────────────────┘
 func (t *Mimir) initQuerier() (serv services.Service, err error) {
+	t.Cfg.Worker.MaxConcurrentRequests = t.Cfg.Querier.EngineConfig.MaxConcurrent
+	t.Cfg.Worker.QuerySchedulerDiscovery = t.Cfg.QueryScheduler.ServiceDiscovery
```

Comment:
Note to reviewers: moved above to let the new utility function `t.Cfg.Worker.IsFrontendOrSchedulerConfigured()` to work correctly.